core/psh/nslookup.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Phoenix-RTOS
+ *
+ * nslookup - make DNS querries
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Maciej Purski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <errno.h>
+
+
+void psh_nslookupinfo(void)
+{
+	printf("  nslookup - queries domain name servers\n");
+}
+
+
+static uint32_t *psh_aiToAddr(struct addrinfo *ai)
+{
+	switch (ai->ai_family) {
+		case AF_INET:
+			return (uint32_t *) &((struct sockaddr_in *) ai->ai_addr)->sin_addr;
+		case AF_INET6:
+			return (uint32_t *) &((struct sockaddr_in6 *) ai->ai_addr)->sin6_addr;
+		default:
+			return NULL;
+	}
+}
+
+
+int psh_nslookup(int argc, char **argv)
+{
+	struct addrinfo *res, *cur;
+	struct addrinfo hints = { };
+	char hostaddr[INET6_ADDRSTRLEN] = { };
+
+	hints.ai_socktype = SOCK_STREAM;
+	hints.ai_flags = AI_CANONNAME;
+
+	if (argc < 2) {
+		fprintf(stderr, "nslookup: hostname required!\n");
+		return EINVAL;
+	}
+
+	if (getaddrinfo(argv[1], NULL, &hints, &res) != 0) {
+		fprintf(stderr, "nslookup: Can't resolve hostname!\n");
+		return EINVAL;
+	}
+
+	cur = res;
+	while (cur) {
+		if (inet_ntop(cur->ai_family, psh_aiToAddr(res), hostaddr, sizeof(hostaddr)) == NULL) {
```

Comment:
Shouldn't `psh_aiToAddr(res)` take `cur` ? And if so, is the `res` variable then necessary.
```suggestion
		if (inet_ntop(cur->ai_family, psh_aiToAddr(cur), hostaddr, sizeof(hostaddr)) == NULL) {
```

Reply:
Yes, it should. You are right.